discord/bot.py

```diff
@@ -208,10 +209,48 @@ def on_stats_command(self, event, args):
             ' for **' + args.date.format('MMMM D, YYYY') + '**' if args.date else ''
         ), embed=utils.create_player_message_embed(player))
 
+    @Plugin.command('evolution', parser=True)
+    @Plugin.parser.add_argument('username')
+    @Plugin.parser.add_argument('type', choices=constants.VALID_EVOLUTION_TYPE_NAMES)
+    @Plugin.parser.add_argument('database', choices=rwr.constants.VALID_DATABASES, nargs='?', default='invasion')
+    def on_evolution_command(self, event, args):
+        """Displays the evolution of the specified stat."""
+        args.username = utils.prepare_username(args.username)
+
+        player = self.rwr_scraper.search_player_by_username(args.database, args.username)
+
+        if not player:
+            event.msg.reply('Sorry dude, this player don\'t exist :confused:')
+
+            return
+
+        if not player.rwr_account:
+            event.msg.reply('Sorry my friend, evolution is not available for this player :confused: He/she must be part of the {} {} most experienced players.'.format(
+                rwr.utils.get_database_name(args.database),
+                app.config['MAX_NUM_OF_PLAYERS_TO_TRACK_STATS_FOR'])
+            )
+
+            return
+
+        player_evolution_data = RwrAccountStat.get_stats_by_column(player.rwr_account.id, constants.VALID_EVOLUTION_TYPES[args.type]['column'])
```

Comment:
Put all the data retrieving and image creation logic in a method, return the image as a result so it can be cached

Reply:
`get_stats_for_column`